--- /home/bgomoluch/racktables/RackTables-0.19.12/gateways/deviceconfig/ios12.connector	2012-03-14 13:58:35.000000000 +0100
+++ ios12.connector	2012-05-21 12:46:01.372473660 +0200
@@ -72,6 +72,14 @@
 	printf 'term len 0\nshow mac address-table dynamic\n' >> "$SESSION"
 	outfile="$WORKFILE"
 	;;
+getuptime)
+        printf 'show version | in uptime\n' >> "$SESSION"
+        outfile="$WORKFILE"
+        ;;
+getinterfaces)
+        printf 'term len 0\nshow interfaces\n' >> "$SESSION"
+        outfile="$WORKFILE"
+        ;;
 deploy)
 	cat "$WORKFILE" >> "$SESSION"
 	outfile=/dev/null
